catalyst_voices/packages/internal/catalyst_voices_models/lib/src/document/signed_document_data.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:equatable/equatable.dart';
+
+// List of types and metadata fields is here
+// https://input-output-hk.github.io/catalyst-libs/branch/feat_signed_object/architecture/08_concepts/signed_doc/types/
+
+extension type const SignedDocumentDataPayload(Map<String, dynamic> data)
+    implements Object {}
+
+enum SignedDocumentType {
+  proposalDocument(uuid: '7808d2ba-d511-40af-84e8-c0d1625fdfdc'),
+  proposalTemplate(uuid: '0ce8ab38-9258-4fbc-a62e-7faa6e58318f');
+
+  const SignedDocumentType({
+    required this.uuid,
+  });
+
+  final String uuid;
```

Comment:
fields go above constructors

```suggestion
   final String uuid;
    
  const SignedDocumentType({
    required this.uuid,
  });
```